Code/Framework/AzToolsFramework/Tests/UI/EntityOutlinerTests.cpp

```diff
@@ -82,8 +82,8 @@ namespace UnitTest
             }
 
             auto transform = aznew AzToolsFramework::Components::TransformComponent;
-            transform->SetParent(parentId);
             entity->AddComponent(transform);
+            transform->SetParent(parentId);
```

Comment:
Nice job tracking this down - can you elaborate on why the order of operations started mattering here?